syncd/SaiSwitchInterface.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+extern "C" {
+#include "sai.h"
+#include "saimetadata.h"
+}
+
+#include <set>
+#include <unordered_map>
+#include <map>
+
+namespace syncd
+{
+    class SaiSwitchInterface
+    {
+        private:
+
+            SaiSwitchInterface(const SaiSwitchInterface&);
+            SaiSwitchInterface& operator=(const SaiSwitchInterface&);
+
+        public:
+
+            SaiSwitchInterface(
+                    _In_ sai_object_id_t switchVid,
+                    _In_ sai_object_id_t switchRid);
+
+            virtual ~SaiSwitchInterface() = default;
+
+        public:
+
+            sai_object_id_t getVid() const;
+            sai_object_id_t getRid() const;
+
+        public:
```

Comment:
>public: [](start = 8, length = 7)

Wondering why you need consequent `public:`

Reply:
i use this as separator for similar blocks, here block of non virtual public functions , and next block of public virtual functions